azafea/event_processors/endless/metrics/v3/migrations/24a2bb2bf13e_materialized_views_for_v2_to_v3.py

```diff
@@ -0,0 +1,251 @@
+# type: ignore
+
+"""Materialized views for metrics v2 and v3 events
+
+Revision ID: 24a2bb2bf13e
+Revises: a82a26763a82
+Create Date: 2021-07-15 16:01:40.617855
+
+"""
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = '24a2bb2bf13e'
+down_revision = 'a82a26763a82'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.execute(
+        "CREATE MATERIALIZED VIEW launched_equivalent_existing_flatpak_view "
+        "(occured_at, os_version, replacement_app_id, argv, image_id, site, dual_boot, live) AS "
+        "SELECT occured_at, os_version, replacement_app_id, argv, image_id, site, dual_boot, live "
+        "FROM launched_equivalent_existing_flatpak_v3 l "
+        "JOIN channel_v3 c ON l.channel_id=c.id "
+        "UNION "
```

Comment:
This should be

```suggestion
        "UNION ALL "
```

`UNION` removes duplicates from the combined data set, and involves extra computation to do so. But we don't want to remove duplicates. In the unlikely event that a v2 and v3 machine records the same event at the same time, we want both in the view. So we should use `UNION ALL` to do less computation and also give the right result.